src/adapter/graphql/src/queries/accounts/accounts.rs

```diff
@@ -39,6 +91,38 @@ impl Accounts {
         let maybe_account = account_service.account_by_name(&account_name).await?;
         Ok(maybe_account.map(Account::from_account))
     }
+
+    /// Returns accounts by their names
+    #[tracing::instrument(level = "info", name = Accounts_by_names, skip_all, fields(?account_names, ?skip_missing))]
+    async fn by_names(
+        &self,
+        ctx: &Context<'_>,
+        account_names: Vec<AccountName<'_>>,
+        #[graphql(
+            desc = "Whether to skip unresolved accounts or return an error if one or more are \
+                    missing"
+        )]
+        skip_missing: bool,
+    ) -> Result<Vec<Account>> {
+        let account_service = from_catalog_n!(ctx, dyn kamu_accounts::AccountService);
+
+        let mut res = Vec::new();
+
+        for account_name in account_names {
+            let account_name = account_name.into();
+
+            // TODO: PERF: Vectorize resolution
+            match account_service.account_by_name(&account_name).await? {
+                Some(account) => res.push(Account::from_account(account)),
+                None if skip_missing => (),
+                None => {
+                    return Err(GqlError::gql(format!("Unresolved account: {account_name}")));
+                }
+            }
+        }
+
```

Comment:
It's probably easy to make a similar vectorized routine down to SQLX queries with minimal changes.